apps/labs/posts/dask-array-api-scipy.md

```diff
@@ -0,0 +1,225 @@
+compute nodes.
+
+## Supporting `dask.array` within scipy
+
+Like many other array libraries (e.g. PyTorch), `dask.array` is not fully array API compatible out of the box.
+To work around these differences, scipy uses the [array-api-compat](https://github.com/data-apis/array-api-compat)
+libary, which acts as a portability layer that wraps functions in various array namespaces to make them array API
+compatible.
+
+The wrapping done for dask can be found [here](https://github.com/data-apis/array-api-compat/pull/76).
+While this resolved many incompabilities of existing `dask.array` functions with the array API standard, we were still
+unable to implement some array API functionality since some functions (mostly in the linear algebra extension)
+were unimplemented by dask.
+(a list of failures can be found [here](https://github.com/data-apis/array-api-compat/blob/main/dask-xfails.txt))
+
+With this, we were now ready to begin testing of dask arrays in scipy, which turned out to be a mixed bag initially.
+While some functions worked perfectly with zero code change, others crashed completely due to incompabilities with
+dask's lazy design or resulted in wrong answers.
+
+After going through the failures, we were able to categorize them into 3 main areas.
+
+### Dask failure modes
+
+1. Data-dependent output shapes
+
+   For the uninitiated [data-dependent output shapes](https://data-apis.org/array-api/latest/design_topics/data_dependent_output_shapes.html)
+   occur when the shape of the resulting array from an array API operation is unknown, because it depends on the
+   values of an input array.
+
+   For example, when calling the [`unique_values`](https://data-apis.org/array-api/latest/API_specification/generated/array_api.unique_values.html)
+   function on an input array, the length of the output array will be dependent on the number of unique values appearing
+   in the input array. Another common case of data-dependent output shapes, and the one we see the most in scipy, comes
+   from the indexing with a boolean mask in numpy. When writing idiomatic numpy code, it is common to use a boolean mask
+   for selecting a subset of items via `__getitem__` (i.e. `x[mask]`), or to set items according to the mask (i.e. `x[mask] = y`).
+   Because the number of items to get/sets depends on the number of True values in the mask, this is a data-dependent operation.
+
+   While some libraries like numpy or PyTorch can handle data-dependent output shapes, jitted/lazy libraries such as dask,
+   jax, or MLX do not support these operations, and it is necessary to rewrite functions to avoid these operations if
+   possible.
+
+   In scipy, the lack of data-dependent output shapes makes up the majority of failures for dask, so fixing this issue
+   is essential to have good support for dask arrays in scipy.
+
+   ** Avoiding data-dependent output shapes **
```

Comment:
```suggestion
   **Avoiding data-dependent output shapes**
```

Currently, the `*` characters are being rendered, but I'm guessing you intended it to be bold?